apps/greencheck/api/views.py

```diff
@@ -97,15 +97,15 @@ def get(self, request, ip_to_check=None, format=None):
 
 
 class ProviderSharedSecretView(views.APIView):
+    # TODO: come up with a better solution to identify a provider in this API
     serializer_class = ProviderSharedSecretSerializer
     authentication_classes = [SessionAuthentication, BasicAuthentication]
-    permission_classes = [BelongsToHostingProvider]
 
     @swagger_auto_schema(tags=["Provider Shared Secret"])
     def get(self, request, format=None):
-        try:
-            provider = request.user.hostingprovider
-        except AttributeError:
+        # implicitly choose a first provider that the user has permissions to manage
```

Comment:
Would you please add a line saying something like `TODO: revisit when updating API with revised permissions in production`? We'll definitely need to come back to this, but I want to make  it clear to anyone looking to integrate with us that we are planning to support multiple providers in future.

Reply:
There's a TODO item a couple of lines above: https://github.com/thegreenwebfoundation/admin-portal/pull/481/files#diff-1509052d177b4feeced011bc5b0c2a26c9837bf7f33abfb09beb26f6b2272640R100